app/task_routes.py

```diff
@@ -0,0 +1,134 @@
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from flask import Blueprint, jsonify, make_response, abort, request
+import datetime
+import requests
+import os
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix = "/tasks")
+
+###### Create Route ######
+@tasks_bp.route("", methods=["POST"])
+def create_task():
+    request_body = request.get_json()
+
+    if not request_body.get("title") or not request_body.get("description"):
+        abort(make_response(
+                {
+                    "details": "Invalid data"
+                }, 400
+            ))
+
+    task = Task.from_dict(request_body)
+
+    db.session.add(task)
+    db.session.commit()
+
+    return jsonify({"task": task.to_dict()}), 201
+
+###### Read Route ######
+
+@tasks_bp.route("", methods=["GET"])
+def read_tasks():
+    
+    title_query = request.args.get("sort")
+    if title_query == "asc":
+        tasks = Task.query.order_by(Task.title).all()
+    elif title_query == "desc":
+        tasks = Task.query.order_by(Task.title.desc()).all()
+    else:
+        tasks = Task.query.all()
+
+    tasks_response = []
+    for task in tasks:
+        tasks_response.append(task.to_dict()), 200
+
+    return jsonify(tasks_response)
+
+@tasks_bp.route("/<task_id>", methods=["GET"])
+def read_one_saved_task(task_id):
+    task = validate_model(Task, task_id)
+
+    return {"task": task.to_dict()}
+    
+###### Update Route ######
+@tasks_bp.route("/<task_id>", methods=["PUT"])
+def update_task(task_id):
+    task = validate_model(Task, task_id)
+
+    request_body = request.get_json()
+
+    task.title = request_body["title"]
+    task.description = request_body["description"]
```

Comment:
Right now, if a user sends a request without the keys title or description your server would crash. There's a couple of ways to handle this, you could call the `validate_request` function before you access the keys in `request_body` or you could implement a `try/except` block.